src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -72,32 +73,72 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
         }
 
         var selectBuilder = GetSelectBuilderForDocumentTable(tablePrefix, providerName);
-
         try
         {
             var selectCommand = connection.CreateCommand();
             selectCommand.CommandText = selectBuilder.ToSqlString();
 
             using var result = await selectCommand.ExecuteReaderAsync();
-
-            // at this point the query succeeded and the table exists
-            return DbConnectionValidatorResult.DocumentFound;
+            if (!isDefaultShell)
```

Comment:
I don't know why default is treated differently here?

Reply:
Because finally we want to check if the ShellDescriptor exists only for the Default tenant. Because for other tenants we will be able to remove them that will remove their Document table.

So for non Default tenant we only check that the Document table exists, otherwise when removing a given tenant we may remove the Document table also used by an app level service, this is only allowed for the Default tenant that we will never be able to remove.